test/base/extendObservable.js

```diff
@@ -0,0 +1,119 @@
+// @ts-check
+
+import {
+    computed,
+    autorun,
+    isObservable,
+    isObservableProp,
+    isComputedProp,
+    extendObservable
+} from "../../src/mobx"
+
+test("extendObservable should work", function() {
+    class Box {
+        // @ts-ignore
+        uninitialized
+        height = 20
+        sizes = [2]
+        someFunc = function() {
+            return 2
+        }
+        get width() {
+            return (
+                this.undeclared *
+                this.height *
+                this.sizes.length *
+                this.someFunc() *
+                (this.uninitialized ? 2 : 1)
+            )
+        }
+        addSize() {
+            // @ts-ignore
+            this.sizes.push([3])
+            // @ts-ignore
+            this.sizes.push([4])
+        }
+        constructor() {
+            this.undeclared = 1
+        }
+    }
+
+    const box = new Box()
+
+    extendObservable(box, {
+        height: 20,
+        sizes: [2],
+        get someFunc() {
+            return 2
+        },
+        width: 40
+    })
+
+    expect(isObservableProp(box, "height")).toBe(true)
+    expect(isObservableProp(box, "sizes")).toBe(true)
+    expect(isObservable(box.sizes)).toBe(true)
+    expect(isObservableProp(box, "someFunc")).toBe(true)
+    expect(isComputedProp(box, "someFunc")).toBe(true)
+    expect(isObservableProp(box, "width")).toBe(true)
```

Comment:
Maybe overwritting a getter with an observable should just throw, as it is probably a mistake? Not sure

Reply:
So something like:

```
if (Object.getOwnPropertyDescriptor(target, key) && Object.getOwnPropertyDescriptor(target, key).get) {
    fail()
}
```
?

Reply:
Yeah precisely that, if you agree with the constraint :)

Reply:
On one hand, I kind of like an all or nothing approach with not special cases like this, but it is unlikely that overwriting a getter is what the user wants to do.

Reply:
Hmm, but what if you want to overwrite a getter with a computed?

Reply:
I think I would leave it with no constraint, and leave it up to the user to understand what they are overwriting.

Reply:
Perfect, makes sense!